cardano-api/test/Test/Cardano/Api/Typed/CBOR.hs

```diff
@@ -5,62 +5,35 @@ module Test.Cardano.Api.Typed.CBOR
   ( tests
   ) where
 
-import           Cardano.Api
-import           Gen.Cardano.Api.Typed
+import           Cardano.Prelude
+
 import           Gen.Hedgehog.Roundtrip.CBOR (roundtrip_CBOR)
-import           Gen.Tasty.Hedgehog.Group (fromGroup)
-import           Hedgehog (Property, discover)
+import           Hedgehog (Property)
 import           Test.Cardano.Api.Typed.Orphans ()
-import           Test.Tasty (TestTree)
+import           Test.Tasty (TestTree, testGroup)
+import           Test.Tasty.Hedgehog (testProperty)
+import           Test.Tasty.TH (testGroupGenerator)
+
+import           Cardano.Api
+import           Gen.Cardano.Api.Typed
 
 {- HLINT ignore "Use camelCase" -}
 
 -- TODO: Need to add PaymentExtendedKey roundtrip tests however
 -- we can't derive an Eq instance for Crypto.HD.XPrv
 
-prop_roundtrip_txbody_byron_CBOR :: Property
-prop_roundtrip_txbody_byron_CBOR =
-  roundtrip_CBOR (AsTxBody AsByronEra) (genTxBody ByronEra)
-
-prop_roundtrip_txbody_shelley_CBOR :: Property
-prop_roundtrip_txbody_shelley_CBOR =
-  roundtrip_CBOR (AsTxBody AsShelleyEra) (genTxBody ShelleyEra)
-
-prop_roundtrip_txbody_allegra_CBOR :: Property
-prop_roundtrip_txbody_allegra_CBOR =
-  roundtrip_CBOR (AsTxBody AsAllegraEra) (genTxBody AllegraEra)
-
-prop_roundtrip_txbody_mary_CBOR :: Property
-prop_roundtrip_txbody_mary_CBOR =
-  roundtrip_CBOR (AsTxBody AsMaryEra) (genTxBody MaryEra)
-
-prop_roundtrip_txbody_alonzo_CBOR :: Property
-prop_roundtrip_txbody_alonzo_CBOR =
-  roundtrip_CBOR (AsTxBody AsAlonzoEra) (genTxBody AlonzoEra)
-
-prop_roundtrip_tx_byron_CBOR :: Property
-prop_roundtrip_tx_byron_CBOR =
-  roundtrip_CBOR (AsTx AsByronEra) (genTx ByronEra)
-
-prop_roundtrip_tx_shelley_CBOR :: Property
-prop_roundtrip_tx_shelley_CBOR =
-  roundtrip_CBOR (AsTx AsShelleyEra) (genTx ShelleyEra)
-
-prop_roundtrip_tx_allegra_CBOR :: Property
-prop_roundtrip_tx_allegra_CBOR =
-  roundtrip_CBOR (AsTx AsAllegraEra) (genTx AllegraEra)
-
-prop_roundtrip_tx_mary_CBOR :: Property
-prop_roundtrip_tx_mary_CBOR =
-  roundtrip_CBOR (AsTx AsMaryEra) (genTx MaryEra)
-
-prop_roundtrip_tx_alonzo_CBOR :: Property
-prop_roundtrip_tx_alonzo_CBOR =
-  roundtrip_CBOR (AsTx AsAlonzoEra) (genTx AlonzoEra)
+test_roundtrip_txbody_CBOR :: [TestTree]
+test_roundtrip_txbody_CBOR =
+  [ testProperty (show era) $
+    roundtrip_CBOR (proxyToAsType Proxy) (genTxBody era)
+  | AnyCardanoEra era <- allCardanoEras
```

Comment:
Yes, indeed better and shorter. Use the `[minBound..maxBound]` idiom here.

Reply:
I've done this mainly to keep it working when new eras are added.